giving-example/src/main/java/com/adyen/giving/service/DonationService.java

```diff
@@ -0,0 +1,41 @@
+package com.adyen.giving.service;
+
+import jakarta.servlet.http.HttpSession;
+import jakarta.ws.rs.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class DonationService {
+    @Autowired
+    protected HttpSession session;
```

Comment:
A Service class should contain business login and avoid a dependency on the `HttpSession` (being agnostic to the layer/protocol invoking it). As there is no business logic in the class maybe we could create a Util class (in the Controller layer) or just a separate private method in `CheckoutResource`

Reply:
I'll create the util class and fix the E2E test